docs/painless/painless-guide/painless-execute-script.asciidoc

```diff
@@ -1,37 +1,137 @@
-
-The Painless execute API allows an arbitrary script to be executed and a result to be returned.
-
-[[painless-execute-api-parameters]]
-.Parameters
-[options="header"]
-|======
-| Name              | Required  | Default                | Description
-| `script`          | yes       | -                      | The script to execute.
-| `context`         | no        | `painless_test`        | The context the script should be executed in.
-| `context_setup`   | no        | -                      | Additional parameters to the context.
-|======
-
-==== Contexts
-
-Contexts control how scripts are executed, what variables are available at runtime and what the return type is.
-
-===== Painless test context
-
-The `painless_test` context executes scripts as is and does not add any special parameters.
-The only variable that is available is `params`, which can be used to access user defined values.
-The result of the script is always converted to a string.
-If no context is specified then this context is used by default.
-
-*Example*
-
-Request:
+[[painless-execute-runtime-context]]
+.Runtime fields context
+[%collapsible%open]
+=====
+--
+The following contexts are specific to runtime fields.
+
+NOTE: Result ordering in the runtime field context is not guaranteed.
+--
+
+****
+`boolean_field`:: 
+Emits a `boolean` data type and returns a sorted list of `boolean` values. See
+<<painless-runtime-boolean,boolean_field>>.
+
+`date_field`:: 
+Emits a `long` data type and returns a sorted list of `long` values. See
+<<painless-runtime-datetime,date_time>>.
+
+`double_field`::
+Emits a `long` data type and returns a sorted list of `double` values. See
+<<painless-runtime-double,double_field>>.
+
+`geo_point_field`::
+Emits a `long` data type and returns a sorted list of `double` values. See
+<<painless-runtime-geo,geo_point_field>>.
+
+`ip_field`::
+Emits a `string` data type and returns a sorted list of `string` values. See
+<<painless-runtime-ip,ip_field>>.
+
+`keyword_field`::
+Emits a `string` data type and returns a sorted list of `string` values. See
+<<painless-runtime-keyword,keyword_field>>.
+
+`long_field`::
+Emits a `long` data type and returns a sorted list of `long` values. See
+<<painless-runtime-long,long_field>>.
+****
+=====
+====
+
+`context_setup`:: (Required, object)
+Additional parameters for the `context`.
++
+NOTE: This parameter is required for all contexts except `painless_test`,
+which is the default if no value is provided for `context`.
++
+.Properties of `context_setup`
+[%collapsible%open]
+====
+`document`:: (Required, string)
+Document that's temporarily indexed in-memory and accessible from the script.
+
+`index`:: (Required, string)
+Index containing a mapping that's compatible with the indexed document.
+====
+
+`params`:: (`Map`, read-only)
+User-defined parameters passed in as part of the query.
+
+`query`:: (Optional, object)
+NOTE: This parameter only applies when `score` is specified as the script 
+`context`.
```

Comment:
Runtime fields don't interact with `_score` so maybe we don't need to talk about it?

Reply:
The `query` parameter isn't specific to the field contexts here, but as a parameter you can include in the overall request body.